server/badge/badge.controller.ts

```diff
@@ -151,15 +151,16 @@ export class BadgeController {
             },
         });
         const parsedUrl = parse(req.url, true);
-        await this.viewService.getNextServer().render(
-            req,
-            res,
-            "/admin-badges",
-            Object.assign(parsedUrl.query, {
-                badges,
-                users,
-                nameSpace: req.params.namespace,
-            })
-        );
+        const query = Object.assign(parsedUrl.query, {
+            badges,
+            users,
+            nameSpace: req.params.namespace,
+        });
+
+        await this.viewService
```

Comment:
suggestion: we should create an abstraction in the viewService that receive the parameters from the controllers but perform the serialization and rendering in a single function